### ISSUE TITLE
Create GIF preview video [2 XMR]

### ISSUE THREAD

**Issue by woodser** [haveno-dex/haveno#1352]
This issue requests creating a brief (~30 seconds) GIF video preview of using Haveno, which can be posted to the official website and/or GitHub repo.

The video must demonstrate using the [official stagenet network](https://github.com/haveno-dex/haveno/blob/master/docs/installing.md), and should include:

- Starting the application
- Creating a payment account
- Posting an offer
- Taking a trade
- Clicking the buttons to confirm payment sent/received
- Showing the trade completion screen
- Briefly showing screens under Funds (maybe make a withdraw tx?)
- Briefly showing screens under Settings
- Briefly showing screens under Account
- Show the Buy > Fiat | Crypto | Other submenus briefly so assets are shown, as well as Sell > Fiat
- Show the "Trade completed successfully" popup, which happens after the close button is clicked for finality.

Please use light mode instead of dark mode.

**Comment by preland** [comment]
I can do this

**Comment by woodser** [comment]
We're actually working on a re-design of part of the UI: https://github.com/haveno-dex/haveno/issues/1351

So it would be best to hold off on this issue until #1351 is complete.

**Comment by preland** [comment]
Ok, sounds good 👍

**Comment by preland** [comment]
I do not think the GIF format would be best for our use case

**Comment by woodser** [comment]
Open to recommendations.

**Comment by TheTollingBell** [comment]
@woodser I'm good to make a `video` and a `gif` of similar quality but I do agree that adding a `gif` to the website will make loading times pretty bad. Additionally, the accepted max length of gifs is usually around 10s.

**Comment by TheTollingBell** [comment]
@woodser Just captured the raw footage. Would you prefer a perspective from seller or buyer or both?

**Comment by woodser** [comment]
Was thinking the video would be hosted on GitHub, so not too concerned with loading time. Great if we can convey most of the flow in ~10 seconds, but overage is ok too. Or maybe we don't use GIF for the final hosted format.

It should be the taker flow, ideally buyer as taker.

**Comment by TheTollingBell** [comment]
Okay, I'm going to edit down the taker flow to around 10sec. I'll post a demo here in about 2hrs hopefully.

**Comment by TheTollingBell** [comment]
@woodser This is the `mp4` for you to review. I had to cut out the majority of the process to fit it within 20 secs. If you want this is definitely convertible to a `gif`. I also have footage on the rest of Haveno such as the account creation process, clicking through the other tabs, and creating a trade.

https://github.com/user-attachments/assets/8104f285-6d6d-4830-8ac9-dc23f374dbd5

**Comment by woodser** [comment]
Thanks @TheTollingBell, this is a good step. Viewing the video, I had these thoughts:

- Screen resolution should be decreased so everything is bigger.
- Show the Buy > Fiat | Crypto | Other menus briefly, and then switch to Sell > Fiat briefly.
- It should show creating a payment account.
- Please show the "Trade completed successfully" popup, which happens after the close button is clicked for finality. This will require the "Do not show" popup settings to be reset which can be done in the application settings.
- It would be nice if additional offers were posted for more context.
- Possible to make the video higher quality, so the text is more legible? Decreasing screen resolution may help.

**Comment by TheTollingBell** [comment]
Okay, I'll have these changes recorded and posted by tonight EST. I'm assuming with these changes that the length of the video is okay to be around 30s-40s. I'm also planning on recording in 1920x1080 this time which should be better.

**Comment by preland** [comment]
I might as well show what I had made; it shows everything in Haveno from starting it up to completing a trade:

https://github.com/user-attachments/assets/1a5297d5-e17f-4210-9a47-92c3a054bfa4

If it appears to have a poor bitrate (or doesn't load in your browser), that's because it is currently encoded with h265 with a high compression rate (I did this so I could quickly upload the video for preview, as the video is 2 and a half minutes long and would otherwise not be uploadable here)

The video can be spliced into smaller clips, potentially with some sections showing in different areas than others, or for use in a tutorial.

**Comment by TheTollingBell** [comment]
Sorry to be off-topic but is that sway-effects or Hyprland, because I love the border config.

**Comment by preland** [comment]
It’s hyprland :)

**Comment by U65535F** [comment]
Since #1351 is completed, this should be no longer postponed right?

**Comment by woodser** [comment]
Correct, just updated the title. Looking for brief clips to set up the account and finish the trade, where any waiting is cut out, to make the video as short as possible (hoping down to ~30-45 seconds or so), highest resolution possible, with compression for reasonable load time.

**Comment by Minecon724** [comment]
Sorry to interrupt, but I made something too:
https://iframe.mediadelivery.net/play/378506/463ac090-98c3-469a-896e-3cc3d6b534b4
The .mkv is just 5 MiB, and that's exported, without any optimizations or heavy compression
Opinions?

**Comment by Minecon724** [comment]
Thanks for reviewing;

> I'm seeing that there's a large black border around the application window. Can the border be removed so the application is full screen in the video?

Of course

> Also ideally the application width is stretched a little more so more content can be displayed. At least the application's default width should be used.

I don't know. Not much is cut off, only the balances (which, having the funds page, aren't that important)
Maybe it should be taller so there's less scrolling?

> And lastly, I think the mouse movements should be slowed down a bit for easier viewing.

That's tough, but I have some ideas

**Comment by Minecon724** [comment]
https://iframe.mediadelivery.net/play/378506/a587e22b-c8d3-48c6-b256-9c071c0c6744

- 58 seconds
- Made the window larger (unlike I thought, it wasn't a terrible idea)
- Didn't apply any speedups, so the mouse movement is a tad slower
- Let's ignore the borders for now

**Comment by woodser** [comment]
Thanks, this is looking very good.

But the black borders are still problematic. It should be possible to utilize the full screen? I think that would work then.

**Comment by Minecon724** [comment]
Fixed: https://iframe.mediadelivery.net/play/378506/2182fa4c-2b36-49c4-9eae-c0ee76c4e20b

It's 1210x716 and it doesn't fit 16/9, but I don't think this is much of an issue?

(I use kdenlive which can be complicated, that's why I didn't want to do it earlier)

**Comment by woodser** [comment]
Any thoughts on @Minecon724's last video submission? https://iframe.mediadelivery.net/play/378506/2182fa4c-2b36-49c4-9eae-c0ee76c4e20b

I was thinking to put this video on our main GitHub readme and maybe add it to haveno.exchange.

It doesn't show the asset pull downs under buy and sell tabs as hoped, but maybe it's good enough for our purposes. Thoughts?

**Comment by Minecon724** [comment]
I uploaded a new version just as you commented (just added the start animation)

> It doesn't show the asset pull downs under buy and sell tabs as hoped,

That's the same list as when I add a payment source, right?

**Comment by woodser** [comment]
> That's the same list as when I add a payment source, isn't it?

Was hoping to show this:

<img width="351" alt="Image" src="https://github.com/user-attachments/assets/9411dd35-3587-4625-bf08-a9d5872358f2" />

<img width="351" alt="Image" src="https://github.com/user-attachments/assets/8b97c1a6-1393-47e3-aabf-0d5179ee1a53" />

**Comment by woodser** [comment]
> just added the start animation

Ah yes that would be good too. I don't see it in the last link though.

**Comment by Minecon724** [comment]
> > That's the same list as when I add a payment source, isn't it?
> 
> Was hoping to show this:

Those are briefly visible as I visit the settings tab

> I don't see it in the last link though.

It's like a quarter second. There's also a fade out at the end to make a loop.

**Comment by woodser** [comment]
Ah yes the startup screen at your latest link: https://iframe.mediadelivery.net/play/378506/674a9eeb-c662-4107-b0a7-b7ed7e431a66

**Comment by woodser** [comment]
Thanks for your video @Minecon724. Some of the feedback we received was that it's trying to squeeze too much in within too little time, which ends up being confusing. I can increase the bounty to 2 XMR for it to be redone with more time, perhaps ~2 minutes instead of 1 minute. It should definitely show the asset dropdowns under the buy/sell tabs, to show what assets can be bought and sold there.

**Comment by Minecon724** [comment]
Good decision to make it longer.
I was about to do it, but I can't take any offers and whenever I post one it gets cancelled, could you help?

**Comment by woodser** [comment]
@Minecon724 That behavior could be expected if building from current master, which necessitates a hard fork. You're building from a tagged release like v1.0.18?

**Comment by Minecon724** [comment]
I was on master. Now I built from 1.0.18 and the issue went away. Thank you for help.

**Comment by Minecon724** [comment]
Here, what do you think?

https://iframe.mediadelivery.net/play/378506/0214eed5-9ba7-4385-be87-6a16a55c716b

**Comment by woodser** [comment]
Thanks @Minecon724. Overall the video is looking good and the pace is about right. I had these minor points of feedback:

- The splash screen appears too quickly (and is maybe sped up?). Ideally it shows at normal playback speed for a couple seconds and then loads.
- Currently it's loading the Settings screen first, and then immediately switches to another screen. It would be better to land on the Market > Offer book view once the application loads, ideally with a basic graph showing, and show this screen for a moment. This is the first screen loaded by default.
- The pace could be slowed down even a little further in some parts, like switching through the Settings screens, but not critical.

I think if the splash screen and first loaded page are fixed, it's pretty much there.

**Comment by Minecon724** [comment]
>  The splash screen appears too quickly (and is maybe sped up?). Ideally it shows at normal playback speed for a couple seconds and then loads.

It is indeed sped up. In my opinion it should be, but you're right that it could be more visible.

Here are some slower versions, please consider:
https://iframe.mediadelivery.net/play/378506/d216c9cd-f793-430d-a9fc-5fb3a2706bf9
https://iframe.mediadelivery.net/play/378506/6c6e714c-604e-4543-80a6-c055e7455a4a
https://iframe.mediadelivery.net/play/378506/84de95df-4117-4d56-8b59-eb819049160a
https://iframe.mediadelivery.net/play/378506/81a081ca-286a-43ff-8043-9830bb0b7eb0



And I'm working on your two other notes!

**Comment by Minecon724** [comment]
Here the startup is not too fast, followed by the chart then buy xmr, and settings are slow enough to pick up a few sentences:

https://iframe.mediadelivery.net/play/378506/2b45815a-dde2-4135-b618-c4633c65be7a

**Comment by woodser** [comment]
> Here the startup is not too fast, followed by the chart then buy xmr, and settings are slow enough to pick up a few sentences:
> 
> https://iframe.mediadelivery.net/play/378506/590fb0d2-15de-46a8-804d-068ad759183c

Hey this looks good, but my only wish is for the Market > Offer book landing page to display longer. Currently it displays for at most 1 second and then switches, but it's such a visual page at the start of the video with much to look at, so I found that jarring. That moment could be used to show the currencies pull down.

Also for the final export, I hope it's possible to export at a higher quality? When reposted to YouTube, etc, they will apply their own compression, which will add more compression artifacts.

Thanks very much for your help.

**Comment by Minecon724** [comment]
How about this?

https://www.youtube.com/watch?v=JmMjFc1bwuo

You can see how it looks like on YT. It's rescaled to 1080p, so that it gets more bitrate. But why would you use youtube? 

Original: https://36ab09b1.m724.eu/hpv3f.mkv

**Comment by woodser** [comment]
Video looks good to me. Thanks :)

**Comment by woodser** [comment]
@Minecon724 Do you mind to upload it to be playable from a website again, for review?

**Comment by woodser** [comment]
I just noticed that the crypto pulldown displays for less than one second, making it difficult to read the available crypto assets. Since showing the available assets is a primary goal, would it be easy to somehow extend this briefly? It's probably not a big deal, but thought I would ask.

**Comment by Minecon724** [comment]
No need to worry, just ask. Here's the fixed version:
https://iframe.mediadelivery.net/play/378506/251f5f2d-bb4c-426b-b1bc-ea3543d3e060

**Comment by woodser** [comment]
This looks good for me. Unless there is any final feedback, I think we can accept @Minecon724 submission and close this issue out.

**Comment by U65535F** [comment]
> This looks good for me. Unless there is any final feedback, I think we can accept [@Minecon724](https://github.com/Minecon724) submission and close this issue out.

@woodser Sure, this looks nice. +1

**Comment by Minecon724** [comment]
https://drive.proton.me/urls/8AF3ZF0Z1W#5rv08xy2yFjf

**Comment by woodser** [comment]
Thanks. Is this the least compressed version possible? It's not bad already, but it's only 18 MBs, so there could be room for growth without problem. If so it's nice to have a larger / less compressed version.

**Comment by Minecon724** [comment]
Yes, it's the best kdenlive allows me to render. Remember, the resolution isn't high and there isn't much going on.

But it could be even smaller, as quality settings in the higher end don't change much. Like here: https://drive.proton.me/urls/15P8Z9QYA4#Lf2kNSKeK9t1

**Comment by Minecon724** [comment]
Here's the kdenlive project: https://drive.proton.me/urls/36EXSAYWMW#7eIYjdJLcbPV

**Comment by woodser** [comment]
Yeah I think the resolution is good enough.

Wish I could open the kdenlive project, but getting white screens for the media:

<img width="664" alt="Image" src="https://github.com/user-attachments/assets/054d4ab1-f384-47ad-9a65-d81f58283a11" />

Anyway, please send or DM an address for the bounty payout. :)

**Comment by Minecon724** [comment]
I was able to import it on another Linux machine.
Maybe it's OS-dependent or something with codecs, I'm not sure. Can you play the clips from "videos/"?

And here's my address: 87wQFyiBZVdbjBL2AdhqhgYpJmZfVNPFBdNmETR1zKNtbPJ1bFjufzZQWvVqv7bsfNBDtALmzZYQwe2PWcxqvVpTHdA1r8H

**Comment by woodser** [comment]
> Can you play the clips from "videos/"?

Yes.

> 87wQFyiBZVdbjBL2AdhqhgYpJmZfVNPFBdNmETR1zKNtbPJ1bFjufzZQWvVqv7bsfNBDtALmzZYQwe2PWcxqvVpTHdA1r8H

Bounty is paid. Thanks for putting this together.

**Comment by Minecon724** [comment]
Received. Thank you very much for the currency.

As for that import issue, I really don't know what could be happening. I can't help you, sorry.

Also, I'll eventually have to delete the files, so save them ASAP.

**Comment by woodser** [comment]
No problem, I've saved the higher res video, lower res video, and project file with assets.

**Comment by woodser** [comment]
@Minecon724 Your video is posted to our [readme](https://github.com/haveno-dex/haveno?tab=readme-ov-file#haveno-demo) and [haveno.exchange](https://haveno.exchange/). :)
